src/App.tsx

```diff
@@ -26,70 +27,129 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+interface State {
+  sortType: SortType,
+  isReversed: boolean,
+}
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  handleSortAlphabetically = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  handleSortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  handleReverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  handleReset = () => {
+    this.setState({ sortType: SortType.NONE, isReversed: false });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const changedGoods = getReorderedGoods(goodsFromServer, this.state);
+    const isGoodsChanged = (sortType !== SortType.NONE) || isReversed;
```

Comment:
```suggestion
    const { sortType, isReversed } = this.state;
    
    const changedGoods = getReorderedGoods(goodsFromServer, this.state);
    const isGoodsChanged = (sortType !== SortType.NONE) || isReversed;
```
it's better to separate destructuring and regular variables